### PR TITLE
Clickup tasks

### PR DESCRIPTION
**Pull request by BHIIKTOR**
* Added Grant and Distribution modules msgs
* Added verification for contract admin is none
* Minor clippy fixes

### CODE REVIEW

**Review by CyberHoward — CHANGES_REQUESTED** [review]
Hey, looks good, you can leave out `Staking` as it is already included in CosmWasm's api. 
Just a small comment about feature-flagging imports

Also you should add exports for these apis in the lib.rs

**Review by CyberHoward — CHANGES_REQUESTED** [review]

File: packages/abstract-sdk/Cargo.toml [hunk 1]
```diff
@@ -23,9 +27,9 @@ cw-controllers = { workspace = true }
 cw-storage-plus = { workspace = true }
 cw2 = { workspace = true }
 cw20 = { workspace = true }
-cosmos-sdk-proto = { git = "https://github.com/cosmos/cosmos-rust.git" }
-prost = "0.11"
-prost-types = "0.11"
+cosmos-sdk-proto = { git = "https://github.com/cosmos/cosmos-rust.git", optional = true }
```
[the remark this answers is not in the document]
Comment (BHIIKTOR) [reply packages/abstract-sdk/Cargo.toml]: yes there is, fixed it

**Comment by CyberHoward** [comment]
> Hey, looks good, you can leave out `Staking` as it is already included in CosmWasm's api. Just a small comment about feature-flagging imports
> 
> Also you should add exports for these apis in the lib.rs

Could you export the apis in lib and fix the clippy warnings?

File: packages/abstract-sdk/Cargo.toml [hunk 2]
```diff
@@ -11,7 +11,7 @@ resolver = "2"
 
 # for quicker tests, cargo test --lib
 [features]
-default = []
```
Comment (CyberHoward) [inline packages/abstract-sdk/Cargo.toml]: We don't want to default this

File: packages/abstract-sdk/src/lib.rs [hunk 3]
```diff
@@ -27,9 +27,12 @@ pub use crate::apis::{
     version_registry::*,
 };
 
+#[cfg(feature = "stargaze")]
+pub use crate::apis::{grant::*, distribution::*};
```
Comment (CyberHoward) [inline packages/abstract-sdk/src/lib.rs]: sweeeet

File: packages/abstract-sdk/Cargo.toml (hunk 2 above)
Comment (BHIIKTOR) [reply packages/abstract-sdk/Cargo.toml]: yea sorry i left it on again...

File: packages/abstract-sdk/src/apis/distribution.rs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    traits::Message,
+};
+use cosmwasm_std::{to_binary, Addr, Coin, CosmosMsg};
+
+use crate::{
+    features::{AbstractNameService, AccountIdentification},
+    AbstractSdkResult,
+};
+
+pub trait DistributionInterface: AbstractNameService + AccountIdentification {
```
Comment (CyberHoward) [inline packages/abstract-sdk/src/apis/distribution.rs:16]: We have these trait bounds here but we're not using any of these traits in the implementation. 

Ideally we already construct the execution message to be called on the proxy contract (see the other apis)

**Review by CyberHoward — CHANGES_REQUESTED** [review]

File: contracts/native/version-control/src/commands.rs [hunk 5]
```diff
@@ -64,6 +64,17 @@ pub fn propose_modules(
             // Only owner can add modules
             validate_account_owner(deps.as_ref(), &module.namespace, &msg_info.sender)?;
         }
+
+        // verify contract admin is None
+        if !deps
```
Comment (CyberHoward) [inline contracts/native/version-control/src/commands.rs]: We only want to check this if the module is an `Adapter`

**Review by CyberHoward — CHANGES_REQUESTED** [review]

File: packages/abstract-sdk/Cargo.toml (hunk 2 above)
Comment (CyberHoward) [reply packages/abstract-sdk/Cargo.toml]: You don't have to turn it on, you can enable the --all-features in your IDE so it will analyze everything

Comment (BHIIKTOR) [reply packages/abstract-sdk/Cargo.toml]: thanks for the tip!

**Review by CyberHoward — APPROVED** [review]

**Pull request merged**